pkg/jaeger/query/query.go

```diff
@@ -41,47 +45,85 @@ func (p *Query) SpanWriter() spanstore.Writer {
 }
 
 func (p *Query) GetTrace(ctx context.Context, traceID model.TraceID) (*model.Trace, error) {
+	metrics.RequestsTotal.With(prometheus.Labels{"type": "trace", "handler": "get_trace", "code": ""}).Inc()
+	start := time.Now()
 	res, err := getTrace(ctx, p.conn, traceID)
-	if err == nil {
-		traceRequestsExec.Add(1)
+	if err != nil {
+		metrics.RequestsTotal.With(prometheus.Labels{"type": "trace", "handler": "get_trace", "code": "500"}).Inc()
```

Comment:
```suggestion
		metrics.RequestsTotal.With(prometheus.Labels{"type": "trace", "handler": "get_trace", "code": "5xx"}).Inc()
```

We are not sure this is exactly `500` (Internal Server Error), but we can be sure this is something from 5xx range (server errors).

---
If you are 100% sure this is always HTTP500 error, then disregard this comment.